src/main/java/org/sagebionetworks/web/client/widget/discussion/modal/NewDiscussionThreadModal.java

```diff
@@ -71,25 +70,37 @@ public void onSave() {
 		discussionForumClient.createThread(toCreate, new AsyncCallback<DiscussionThreadBundle>(){
 			@Override
 			public void onFailure(Throwable caught) {
-				// TODO Auto-generated method stub
-				
+				synAlert.handleException(caught);
 			}
 
 			@Override
 			public void onSuccess(DiscussionThreadBundle result) {
+				view.hideDialog();
+				view.showSuccess();
 				if (newThreadCallback != null) {
-					newThreadCallback.invoke(null);
+					newThreadCallback.invoke();
 				}
 			}
 		});
 	}
 
-	private boolean isValidMessage(String messageMarkdown) {
-		return (messageMarkdown != null && !messageMarkdown.equals(""));
-	}
-
-	private boolean isValidTitle(String threadTitle) {
-		return (threadTitle != null && !threadTitle.equals(""));
+	public static ValidationResult validate(String threadTitle, String messageMarkdown) {
+		String errorMessages = "";
+		ValidationResult result = new ValidationResult();
+		result.setValidity(true);
+		if (threadTitle == null || threadTitle.equals("")) {
```

Comment:
I like this idea, and think it could be used across the app.  
What do you think about adding a new method to ValidationResult:
```
ValidationResult requiredField(String fieldName, String fieldValue)
```
My suggestion uses <a href='https://en.wikipedia.org/wiki/Method_chaining'>method chaining</a>, so that in your case validate() would become:
```
ValidationResult result = new ValidationResult();
result.requiredField("Title", threadTitle)
      .requiredField("Message", messageMarkdown);
return result;
```